components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -73,6 +88,10 @@ macro_rules! impl_primitive_trie_value {
             fn try_from_u32(i: u32) -> Result<Self, Self::TryFromU32Error> {
                 Self::try_from(i)
             }
+
+            fn to_u32(self) -> u32 {
+                u32::try_from(self).unwrap_or(0)
```

Comment:
q: If we remove the unsigned types from here, we can avoid the GIGO. I don't think TrieValue should ever be of a type that _cannot_ be fully mapped to u32. Thoughts?

Reply:
It's the same issue as when we encode signed ints to RawBytesULE. I'd say just convert the signed numbers to i32 and cast the bytes to u32.

Reply:
Nit: IMO, this should map `-1` to `u32::MAX`, etc. (cast the bits).

Reply:
That's harder to do with the macro, I could split it out

Reply:
I think `self as u32` does the bidding, no?

Reply:
Oh yeah that works now (it used to be UB)